megalodon/src/entities/account.ts

```diff
@@ -1,7 +1,7 @@
-import { Emoji } from './emoji'
-import { Field } from './field'
-import { Source } from './source'
-import { Role } from './role'
+import { Emoji } from './emoji.js'
+import { Field } from './field.js'
+import { Source } from './source.js'
+import { Role } from './role.js'
```

Comment:
Do we have to specify the extension `.js` for all files?

Reply:
Yup, that's required to have a valid Node ESM project!

It's a consequence of setting `"moduleResolution": "nodenext"` in the tsconfig, which you need for native ESM (no bundler).

When using a bundler, you can use `"moduleResolution": "bundler`, which doesn't force you to use explicit file extensions. However, in modern ESM projects it's usually recommended to always use file extensions in imports (regardless of if strictly necessary) for the sake of consistency between backend/frontend projects. Some enforce it at the linter level: [eslint-plugin-require-extensions](https://www.npmjs.com/package/eslint-plugin-require-extensions).

Reply:
Thank you for the explanation. 
What happens if I forget to add `.js` extension where I am referencing a local file?

Reply:
You would get an explicit TypeScript error telling you the import can't be resolved unless you add ".js".

That would happen at build time (not runtime), so just building the app successfully means your imports would be correct.